### DISCUSSION TITLE
How to configure webpack to create a self containing excalidraw package?

### DISCUSSION THREAD

**Discussion by zsviczian** [excalidraw/excalidraw#3961] [Q&A] [answered]
Excalidraw downloads chunks from https://unpkg.com/. This approach is excellent for web deployment, however, in Obsidian, the user downloads the entire plugin in a main.js file, there is no need to load chunks over the network. I have received user feedback that the plugin does not work in certain offline scenarios.

Because of some tweaks to the "off-the-shelf" Excalidraw, I am now using my own Excalidraw build: https://github.com/zsviczian/excalidraw/tree/for-obsidian/src/packages/excalidraw

But, unfortunately, I am not familiar with webpack and yarn... and so far I wasn't able to figure out what I need to change to build everything, including fonts into a single (local) package. 

I'd appreciate some help with this!

**Comment by zsviczian** [comment, marked as answer]
Solved it by modifying webpack.prod.config.js 

modifying rules for woff2 files:
```js
  module: {
    rules: [
   ....
      {
        test: /\.(woff|woff2|eot|ttf|otf)$/,
        type: 'asset/inline'
      },
```

and adding 

```js
const webpack = require('webpack');
....
plugins: [
new webpack.optimize.LimitChunkCountPlugin({maxChunks: 1})
],
```

**Reply by ad1992** [reply]
@zsviczian yes with specifying max chunksize as `1` you are specifying that only one chunk is created, so files which are lazily loaded will be now part of main bundle. But do you need to inline the fonts as well ? You could remove that and instead preload the fonts so that bundle size is not significantly increased.

**Reply by zsviczian** [reply]
Obsidian downloads plugins to a local folder. By having everything bundled Excalidraw can run in an island mode (which is the case with some of my users). Normally I wouldn't want such a big bundle, but because of the plugin architecture of Obsidian, this is actually preferred.
